library/src/main/java/com/anjlab/android/iab/v3/SkuDetails.java

```diff
@@ -64,6 +79,25 @@ public SkuDetails(JSONObject source) throws JSONException
 		priceLong = source.optLong(Constants.RESPONSE_PRICE_MICROS);
 		priceValue = priceLong / 1000000d;
 		priceText = source.optString(Constants.RESPONSE_PRICE);
+        subscriptionPeriod = parsePeriod(source.optString(Constants.RESPONSE_SUBSCRIPTION_PERIOD, null));
+        subscriptionFreeTrialPeriod = parsePeriod(source.optString(Constants.RESPONSE_FREE_TRIAL_PERIOD, null));
+        introductoryPriceLong = source.optLong(Constants.RESPONSE_INTRODUCTORY_PRICE_MICROS);
+        introductoryPriceValue = introductoryPriceLong / 1000000d;
+        introductoryPriceText = source.optString(Constants.RESPONSE_INTRODUCTORY_PRICE, null);
```

Comment:
There is one solution. Should be some attribute which show that sku haven't trial or introduction period. I can add extra boolean fields for each criteria. I agree that it will be more safety